src/com/carolinarollergirls/scoreboard/defaults/DefaultStatsModel.java

```diff
@@ -0,0 +1,504 @@
+            }
+          }
+        }
+        ts.setTotalScore(t.getScore());
+        ts.setJamScore(t.getScore() - t.getLastScore());
+        ts.setLeadJammer(t.getLeadJammer());
+        ts.setStarPass(t.isStarPass());
+        ts.setTimeouts(t.getTimeouts());
+        ts.setOfficialReviews(t.getOfficialReviews());
+      }
+
+      requestBatchEnd();
+    }
+  };
+
+  protected ScoreBoardListener jamStopListener = new ScoreBoardListener() {
+    public void scoreBoardChange(ScoreBoardEvent event) {
+      requestBatchStart();
+      Clock pc = scoreBoard.getClock(Clock.ID_PERIOD);
+      Clock jc = scoreBoard.getClock(Clock.ID_JAM);
+      JamStatsModel js = getCurentJam();
+      js.setJamClockElapsedEnd(jc.getTimeElapsed());
+      js.setPeriodClockElapsedEnd(pc.getTimeElapsed());
+      js.setPeriodClockWalltimeEnd(System.currentTimeMillis());
+      requestBatchEnd();
+    }
+  };
+
+  protected ScoreBoardListener teamEventListener = new ScoreBoardListener() {
+    public void scoreBoardChange(ScoreBoardEvent event) {
+      requestBatchStart();
+      Clock jc = scoreBoard.getClock(Clock.ID_JAM);
+      Team t = (Team)event.getProvider();
+      TeamStatsModel ts = getCurentJam().getTeamStatsModel(t.getId());
+      ts.setTotalScore(t.getScore());
+      ts.setJamScore(t.getScore() - t.getLastScore());
+      if (jc.isRunning()) {
+        // Only set lead/star pass during a jam, to avoid
+        // resetting it at the end of a jam.
+        ts.setLeadJammer(t.getLeadJammer());
+        ts.setStarPass(t.isStarPass());
+      }
+      ts.setTimeouts(t.getTimeouts());
+      ts.setOfficialReviews(t.getOfficialReviews());
+      requestBatchEnd();
+    }
+  };
+
+  protected ScoreBoardListener skaterEventListener = new ScoreBoardListener() {
+    public void scoreBoardChange(ScoreBoardEvent event) {
+      requestBatchStart();
+      Clock jc = scoreBoard.getClock(Clock.ID_JAM);
+      Skater s = (Skater)event.getProvider();
+      TeamStatsModel ts = getCurentJam().getTeamStatsModel(s.getTeam().getId());
+      if (jc.isRunning()) {
+        // If the jam is over, any skater changes are for the next jam.
+        // We'll catch them when the jam starts.
+        if (s.getPosition() == Position.ID_BENCH) {
+          ts.removeSkaterStatsModel(s.getId());
+        } else {
+          ts.addSkaterStatsModel(s.getId());
+          SkaterStatsModel ssm = ts.getSkaterStatsModel(s.getId());
+          ssm.setPosition(s.getPosition());
+          ssm.setPenaltyBox(s.isPenaltyBox());
+        }
+      }
+      requestBatchEnd();
+    }
+  };
+
+  protected JamStatsModel getCurentJam() {
+    int p = scoreBoard.getClock(Clock.ID_PERIOD).getNumber();
+    int j = scoreBoard.getClock(Clock.ID_JAM).getNumber();
+    ensureAtLeastNPeriods(p);
+    PeriodStatsModel period = getPeriodStatsModel(p);
+    period.ensureAtLeastNJams(j);
+    return period.getJamStatsModel(j);
+  }
+
+  public void ensureAtLeastNPeriods(int n) {
+    synchronized (this) {
```

Comment:
Is there a reason to go with `synchronized (this)` in this file instead of just adding the `synchronized` keyword to the method signature?

Reply:
Done